src/Microsoft.Data/ArrowStringColumn.cs

```diff
@@ -0,0 +1,348 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using Apache.Arrow;
+using Apache.Arrow.Types;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// An immutable to hold Arrow style strings
+    /// </summary>
+    public partial class ArrowStringColumn : BaseColumn
+    {
+        private IList<DataFrameBuffer<byte>> _dataBuffers;
+        private IList<DataFrameBuffer<int>> _offsetsBuffers;
+        private IList<DataFrameBuffer<byte>> _nullBitMapBuffers;
+
+        private readonly Encoding _encoding = Encoding.UTF8;
+
+        // Need a way to differentiate between columns initialized with default values and those with null values in SetValidityBit
+        internal bool _modifyNullCountWhileIndexing = true;
```

Comment:
Is this always `true`? I don't see it being set to false. Also, it shouldn't be exposed outside of the class unless absolutely necessary. And if it is exposed outside of the class, it shouldn't be named with an `_` prefix.